src/lib.rs

```diff
@@ -11,6 +11,898 @@
+                    service_id,
+                    instance_id,
+                    method_id,
+                );
+
+                trace!(
+                    "{}:{} - Unregistered vsomeip message handler.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+
+                Ok(())
+            }
+            RegistrationType::AllPointToPoint(_) => Err(UStatus::fail_with_code(
+                UCode::INTERNAL,
+                "Should be impossible to unregister point-to-point internally",
+            )),
+        }
+    }
+
+    async fn send_internal(
+        umsg: UMessage,
+        application_wrapper: &mut UniquePtr<ApplicationWrapper>,
+        runtime_wrapper: &UniquePtr<RuntimeWrapper>,
+    ) -> Result<(), UStatus> {
+        match umsg
+            .attributes
+            .type_
+            .enum_value_or(UMessageType::UMESSAGE_TYPE_UNSPECIFIED)
+        {
+            UMessageType::UMESSAGE_TYPE_UNSPECIFIED => {
+                return Err(UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Unspecified message type not supported",
+                ));
+            }
+            UMessageType::UMESSAGE_TYPE_NOTIFICATION => {
+                return Err(UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Notification is not supported",
+                ));
+            }
+            UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                let vsomeip_msg_res =
+                    convert_umsg_to_vsomeip_msg(&umsg, application_wrapper, runtime_wrapper).await;
+
+                let Ok(mut vsomeip_msg) = vsomeip_msg_res else {
+                    let err = vsomeip_msg_res.err().unwrap();
+                    error!(
+                        "{}:{} Converting UMessage to vsomeip message failed: {:?}",
+                        UP_CLIENT_VSOMEIP_TAG, UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL, err
+                    );
+                    return Err(err);
+                };
+
+                let service_id = get_pinned_message_base(&vsomeip_msg).get_service();
+                let instance_id = get_pinned_message_base(&vsomeip_msg).get_instance();
+                let event_id = get_pinned_message_base(&vsomeip_msg).get_method();
+                let _interface_version =
+                    get_pinned_message_base(&vsomeip_msg).get_interface_version();
+
+                trace!(
+                    "{}:{} Sending SOME/IP NOTIFICATION with service: {} instance: {} event: {}",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL,
+                    service_id,
+                    instance_id,
+                    event_id
+                );
+
+                let payload = get_message_payload(&mut vsomeip_msg).get_shared_ptr();
+                // TODO: Note that we cannot set the interface_version
```

Comment:
Didn't seem like I can set the interface_version / ue_version_major here. Wondering if y'all know if this is okay @int0x27 / @stevenhartley